iceoryx_examples/icedocker/README.md

```diff
@@ -0,0 +1,105 @@
+                         | iox-roudi |
+                         +-----------+
+
+       +-------------------+        +--------------------+
+       | docker 2          | send   | docker 3           |
+       |                   |------->|                    |
+       | iox-cpp-publisher | data   | iox-cpp-subscriber |
+       +-------------------+        +--------------------+
+```
+
+## Requirements
+
+### Shared Access to Unix Domain Sockets
+
+Every iceoryx application is registering itself at our central broker RouDi
+by sending a message to the unix domain socket located at
+`IOX_UDS_SOCKET_PATH_PREFIX/roudi` which is defined in the corresponding
+platform settings file `platform_settings.hpp`. In linux the socket file handle
+can be found at `/tmp/roudi`. While registering it announces its unix
+domain socket to roudi for the responses of application requests which were
+send to roudi.
+This socket is stored as well in `/tmp/IOX_RUNTIME_NAME`. The `iox-cpp-publisher`
+runtime has the same name as the binary which leads to the socket
+`/tmp/iox-cpp-publisher`.
+
+### Shared Access to File Locks
+
+Iceoryx applications ensure that every runtime name is unique in the system
+by creating a file lock before creating the runtime. This is stored in 
+`IOX_LOCK_FILE_PATH_PREFIX/IOX_RUNTIME_NAME.lock` whereby 
+`IOX_LOCK_FILE_PATH_PREFIX` is defined in the platform settings file
+`platform_settings.hpp`. When running the icedelivery example in a linux
+environment one can observe
+the lock files `/tmp/roudi.lock`, `/tmp/iox-cpp-subscriber.lock` and
+`/tmp/iox-cpp-publisher.lock`.
+
+### Shared Access to Semaphores and Shared Memory
+
+One of the tasks of the central broker RouDi is to create and distribute shared
+memory. When the `iox-cpp-publisher` would like to send data it acquires a
+pointer to this shared memory, writes the data into it and sends the
+pointer to the `iox-cpp-subscriber` which reads the memory at the received
+memory position.
+Additionally, it is possible to signal events across process boundaries via
+semaphores. For instance to signal a subscriber that data has arrived.
+
+## Implementation
+
+To have shared access to the required resources we have to bind the host 
+filesystem:
+
+ * `/tmp`
+ * `/dev`
+
+into every docker container.
+
+### Example
+
+We start in 3 separate terminals 3 docker instances. In this example we
+use `archlinux:latest` but one is free to choose any other linux distribution.
+The iceoryx repository which contains an already build iceoryx can be found at
+`/home/user/iceoryx` which is bound to `/iceoryx`. The usage and building is 
+explained in detail in the 
+[icedelivery C++ example](https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_examples/icedelivery).
+
+#### Terminal 1 (iox-roudi)
+```
+docker run --mount type=bind,source="/dev",target=/dev --mount type=bind,source=/home/user/iceoryx,target=/iceoryx --mount type=bind,source=/tmp,target=/tmp -it archlinux:latest
```

Comment:
nit: use `-v /dev:/dev -v /home/user/iceoryx:/iceoryx -v /tmp:/tmp`, this is less verbose, I agreed, but is widely used.
nit2: you can run command directly, w/o `cd /iceoryx; ./build/iox-roudi`

Finally command can look like
```
docker run -v /dev:/dev -v /home/user/iceoryx:/iceoryx -v /tmp:/tmp archlinux:latest /iceoryx/build/iox-roudi
```

Reply:
@r7vme I would prefer the more verbose way here since I would like to make sure that the user knows what we do behind the scenes.